### PR TITLE
Collection properties following FX guidelines

### PR DESCRIPTION
**Pull request by kirankumarkolli**
Collection properties following FX guidelines

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/ClientOptions.cs [hunk 1]
```diff
@@ -124,23 +124,9 @@ public virtual int GatewayModeMaxConnectionLimit
         /// </summary>
         /// <seealso cref="CosmosClientBuilder.AddCustomHandlers(CosmosRequestHandler[])"/>
         [JsonConverter(typeof(ClientOptionJsonConverter))]
-        public virtual ReadOnlyCollection<CosmosRequestHandler> CustomHandlers
+        public virtual Collection<CosmosRequestHandler> CustomHandlers
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/ClientOptions.cs]: Why remove the read only?

Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/ClientOptions.cs]: Because he is changing the behavior to be appendable?

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/ClientOptions.cs]: Its about who owns the mutability. Before its replaced by new collection instance. 
Now user can only update its contents to include custom headers.

Comment (simplynaveen20) [inline Microsoft.Azure.Cosmos/src/ClientOptions.cs]: What is the ideal behind ReadOnlyCollection  => Collection ? Doesn't it become mutable directly along with AddCustomHandlers on CosmosClientBuilder?

Comment (simplynaveen20) [reply Microsoft.Azure.Cosmos/src/ClientOptions.cs]: Didn't saw the earlier comments on this

Comment (simplynaveen20) [reply Microsoft.Azure.Cosmos/src/ClientOptions.cs]: Doesn't it become mutable directly along with AddCustomHandlers on CosmosClientBuilder?

**Pull request merged**